crates/matrix-sdk/src/sliding_sync/builder.rs

```diff
@@ -109,87 +107,87 @@ impl SlidingSyncBuilder {
     pub fn with_all_extensions(mut self) -> Self {
         {
             let mut cfg = self.extensions.get_or_insert_with(Default::default);
-            if cfg.to_device.is_none() {
-                cfg.to_device = Some(assign!(ToDeviceConfig::default(), { enabled: Some(true) }));
+            if cfg.to_device.is_empty() {
+                cfg.to_device.enabled = Some(true);
             }
 
-            if cfg.e2ee.is_none() {
-                cfg.e2ee = Some(assign!(E2EEConfig::default(), { enabled: Some(true) }));
+            if cfg.e2ee.is_empty() {
+                cfg.e2ee.enabled = Some(true);
             }
 
-            if cfg.account_data.is_none() {
-                cfg.account_data =
-                    Some(assign!(AccountDataConfig::default(), { enabled: Some(true) }));
+            if cfg.account_data.is_empty() {
+                cfg.account_data.enabled = Some(true);
             }
 
-            if cfg.receipts.is_none() {
-                cfg.receipts = Some(assign!(ReceiptsConfig::default(), { enabled: Some(true) }));
+            if cfg.receipts.is_empty() {
+                cfg.receipts.enabled = Some(true);
             }
 
-            if cfg.typing.is_none() {
-                cfg.typing = Some(assign!(TypingConfig::default(), { enabled: Some(true) }));
+            if cfg.typing.is_empty() {
+                cfg.typing.enabled = Some(true);
             }
         }
         self
     }
 
     /// Set the E2EE extension configuration.
     pub fn with_e2ee_extension(mut self, e2ee: E2EEConfig) -> Self {
-        self.extensions.get_or_insert_with(Default::default).e2ee = Some(e2ee);
+        self.extensions.get_or_insert_with(Default::default).e2ee = e2ee;
         self
     }
 
     /// Unset the E2EE extension configuration.
     pub fn without_e2ee_extension(mut self) -> Self {
-        self.extensions.get_or_insert_with(Default::default).e2ee = None;
+        self.extensions.get_or_insert_with(Default::default).e2ee = E2EEConfig::default();
         self
     }
 
     /// Set the ToDevice extension configuration.
     pub fn with_to_device_extension(mut self, to_device: ToDeviceConfig) -> Self {
-        self.extensions.get_or_insert_with(Default::default).to_device = Some(to_device);
+        self.extensions.get_or_insert_with(Default::default).to_device = to_device;
         self
     }
 
     /// Unset the ToDevice extension configuration.
     pub fn without_to_device_extension(mut self) -> Self {
-        self.extensions.get_or_insert_with(Default::default).to_device = None;
+        self.extensions.get_or_insert_with(Default::default).to_device = ToDeviceConfig::default();
```

Comment:
I wonder why this is not explicitly disabling the extension instead. But even if that makes more sense, it should happen in a separate PR. cc @Hywan